lib/ansible/executor/play_iterator.py

```diff
@@ -181,6 +185,8 @@ def __init__(self, inventory, play, play_context, variable_manager, all_vars, st
         }
         if gather_timeout:
             setup_task.args['gather_timeout'] = gather_timeout
+        if fact_path:
+            setup_task.args['fact_path'] = fact_path
```

Comment:
sorry, my comment might be misleadingly placed, i'm not against the 'if/task.args' update, but on how you inject the default here, just use normal play object inheritance upto this point.

Possibly the best place to 'resolve' the default is the task after it checks parents for inheritance. We don't have a clear rule, for example have strategy set this way, but most other defaults are populated in play_context. @jimi-c any opinions on this?